builder/lxc/step_lxc_create.go

```diff
@@ -17,51 +19,83 @@ type stepLxcCreate struct{}
+	if config.PackerForce {
+		s.destroy(name, ui)
 	}
 
-	commands := make([][]string, 3)
-	if config.CloneSource == "" {
-		commands[0] = append(config.EnvVars, []string{"lxc-create", "-n", name, "-t", config.Name, "--"}...)
-		commands[0] = append(commands[0], config.Parameters...)
+	if config.LxcTemplate.Name != "" {
+		commands := make([][]string, 2)
+		commands[0] = append(config.LxcTemplate.EnvVars, []string{"lxc-create", "-n", name, "-t", config.LxcTemplate.Name, "--"}...)
+		commands[0] = append(commands[0], config.LxcTemplate.Parameters...)
+		// prevent tmp from being cleaned on boot, we put provisioning scripts there
+		// TODO: wait for init to finish before moving on to provisioning instead of this
+		commands[1] = []string{"touch", filepath.Join(rootfs, "tmp", ".tmpfs")}
+
+		ui.Say("Creating container from template...")
+		for _, command := range commands {
+			log.Printf("Executing sudo command: %#v", command)
+			if err := s.SudoCommand(command...); err != nil {
+				errorHandler(fmt.Errorf("Command (%s) failed with error: %s", strings.Join(command, " "), err))
+				return multistep.ActionHalt
+			}
+		}
 	} else {
-		commands[0] = []string{"lxc-clone", "--orig", config.CloneSource, "--new", name}
-	}
-	// prevent tmp from being cleaned on boot, we put provisioning scripts there
-	// todo: wait for init to finish before moving on to provisioning instead of this
-	commands[1] = []string{"touch", filepath.Join(rootfs, "tmp", ".tmpfs")}
-	commands[2] = []string{"lxc-start", "-d", "--name", name}
-
-	ui.Say("Creating container...")
-	for _, command := range commands {
-		log.Printf("Executing sudo command: %#v", command)
-		err := s.SudoCommand(command...)
+		containerPath := filepath.Join(lxc_dir, name)
+		containerConfig, err := NewLxcConfig(config.RootFs.ConfigFile)
+		if err != nil {
+			errorHandler(fmt.Errorf("Could not read lxc config (%s): %s", config.RootFs.ConfigFile, err))
+			return multistep.ActionHalt
+		}
+		containerConfig.SetRootFs(rootfs)
+		tmpDir, err := ioutil.TempDir("", "lxcconfig")
 		if err != nil {
-			err := fmt.Errorf("Error creating container: %s", err)
-			state.Put("error", err)
-			ui.Error(err.Error())
+			errorHandler(fmt.Errorf("Could not create temp directory (%s): %s", tmpDir, err))
 			return multistep.ActionHalt
 		}
+		err = containerConfig.Write(filepath.Join(tmpDir, "lxc.config"))
+		if err != nil {
+			os.RemoveAll(tmpDir)
+			errorHandler(fmt.Errorf("Could not write lxc config to %s: %s", filepath.Join(tmpDir, "lxc.config"), err))
+			return multistep.ActionHalt
+		}
+
+		commands := make([][]string, 3)
+		commands[0] = []string{"mkdir", containerPath}
+		commands[1] = []string{"tar", "-C", containerPath, "-xf", config.RootFs.Archive}
+		commands[2] = []string{"cp", filepath.Join(tmpDir, "lxc.config"), filepath.Join(containerPath, "config")}
+
+		ui.Say(fmt.Sprintf("Creating container from archive: %s", config.RootFs.Archive))
+		for _, command := range commands {
```

Comment:
Is this for loop any different from the one called in the other side of the if branch? If not then we can pull it out of the if block or method or both.